src/lib/protocols/mdns/Publisher.h

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "core/CHIPError.h"
+#include "inet/InetInterface.h"
+
+namespace chip {
+namespace Protocols {
+namespace Mdns {
+
+class Publisher
+{
+public:
+    Publisher() = default;
+
+    /**
+     * This method initializes the publisher.
```

Comment:
nit: I would delete the doc commment as it uses vertical space. 

it is obvious that 'Init' initializes and that a CHIP_ERROR return code is NO_ERROR on success and something else on failure.